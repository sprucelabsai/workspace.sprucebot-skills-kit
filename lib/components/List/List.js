'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});
exports.ListItem = exports.List = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Avatar = require('../Avatar/Avatar');

var _Avatar2 = _interopRequireDefault(_Avatar);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var List = exports.List = _styledComponents2.default.div.attrs({
	className: 'item__list'
}).withConfig({
	displayName: 'List',
	componentId: 'g8e11t-0'
})(['']);

var ListItem = exports.ListItem = function (_Component) {
	_inherits(ListItem, _Component);

	function ListItem() {
		_classCallCheck(this, ListItem);

		return _possibleConstructorReturn(this, (ListItem.__proto__ || Object.getPrototypeOf(ListItem)).apply(this, arguments));
	}

	_createClass(ListItem, [{
		key: 'render',
		value: function render() {
			var props = Object.assign({}, this.props);
			var className = props.className,
			    children = props.children,
			    title = props.title,
			    subtitle = props.subtitle,
			    rightInput = props.rightInput,
			    rightTitle = props.rightTitle,
			    rightSubtitle = props.rightSubtitle,
			    online = props.online,
			    avatar = props.avatar;

			// cleanup props for pass through

			delete props.className;
			delete props.children;
			delete props.title;
			delete props.subtitle;
			delete props.rightInput;
			delete props.rightTitle;
			delete props.rightSubtitle;
			delete props.online;
			delete props.avatar;

			// setup class name
			className = (className || '') + ' item__list__item ' + (online ? '' : 'offline');

			// build children
			children = children || [];
			if (!Array.isArray(children)) {
				children = [children];
			}

			// setup title/subtitle
			if (subtitle) {
				children.unshift(_react2.default.createElement(
					'div',
					{ key: 'subtitle', className: 'sub__title' },
					subtitle
				));
			}

			if (title) {
				children.unshift(_react2.default.createElement(
					'div',
					{ key: 'title', className: 'title' },
					title
				));
			}

			return _react2.default.createElement(
				'div',
				_extends({ className: className }, props),
				avatar && _react2.default.createElement(
					'div',
					{ className: 'avatar__outer__wrapper' },
					_react2.default.createElement(_Avatar2.default, { online: online, image: avatar })
				),
				children && _react2.default.createElement(
					'div',
					{ className: 'item__details' },
					children
				),
				rightInput && rightInput,
				(rightTitle || rightSubtitle) && _react2.default.createElement(
					'div',
					{ className: 'content__right' },
					rightTitle && _react2.default.createElement(
						'div',
						{ className: 'title' },
						rightTitle
					),
					rightSubtitle && _react2.default.createElement(
						'div',
						{ className: 'sub__title' },
						rightSubtitle
					)
				)
			);
		}
	}]);

	return ListItem;
}(_react.Component);

ListItem.propTypes = {
	friend: _propTypes2.default.object,
	title: _propTypes2.default.string,
	subtitle: _propTypes2.default.string,
	rightInput: _propTypes2.default.any,
	rightTitle: _propTypes2.default.any,
	rightSubtitle: _propTypes2.default.any,
	online: _propTypes2.default.bool,
	avatar: _propTypes2.default.string
};

ListItem.defaultProps = {
	online: true
};