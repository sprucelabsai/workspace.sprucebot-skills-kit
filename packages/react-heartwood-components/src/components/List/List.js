// @flow
import React, { Fragment } from 'react'
import type { Node } from 'react'
import cx from 'classnames'
import ListHeader from './components/ListHeader/ListHeader'
import ListItem from './components/ListItem/ListItem'
import type { Props as ListHeaderProps } from './components/ListHeader/ListHeader'
import type { Props as ListItemProps } from './components/ListItem/ListItem'

export const ListWrapper = (props: { children: Node }) => (
	<div className="list-wrapper">{props.children}</div>
)

export type Props = {
	/** List Header */
	header?: ListHeaderProps,

	/** List items */
	items?: Array<ListItemProps>,

	/** Class for the list */
	className?: string,

	/** Set true to make the list smaller */
	isSmall?: boolean,

	/** any passthrough to render in the body of the list */
	children?: any,

	/** Set to true to show separators between list items */
	isShowingSeparators: boolean
}

const List = (props: Props) => {
	const {
		header,
		items,
		className,
		isSmall,
		isShowingSeparators,
		children
	} = props
	const parentClass = cx('list', className, {
		'list-small': isSmall,
		'list--separators-hidden': !isShowingSeparators
	})

	return (
		<Fragment>
			{header && <ListHeader isSmall={isSmall} {...header} />}
			<ul className={parentClass}>
				{items && items.map((item, idx) => <ListItem key={idx} {...item} />)}
				{children && children}
			</ul>
		</Fragment>
	)
}

List.defaultProps = {
	header: null,
	className: '',
	isSmall: false,
	isShowingSeparators: true
}

export default List
