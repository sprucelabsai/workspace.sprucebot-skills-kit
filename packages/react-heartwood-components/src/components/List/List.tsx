import React, { Fragment } from 'react'
import cx from 'classnames'
import ListHeader, {
	IListHeaderProps
} from './components/ListHeader/ListHeader'
import ListItem, { IListItemProps } from './components/ListItem/ListItem'
import ExpandableListItem, {
	IExpandableListItemProps
} from './components/ExpandableListItem/ExpandableListItem'
import { IHWList } from '@sprucelabs/spruce-types'

export const ListWrapper = (props): React.ReactElement => (
	<div className="list-wrapper">{props.children}</div>
)

export type IWrappedItemProps = IListItemProps | IExpandableListItemProps

export interface IListProps extends Omit<IHWList, 'id' | 'header' | 'items'> {
	/** optional id for view caching */
	id?: string

	/** List Header */
	header?: IListHeaderProps

	/** List items */
	items?: IWrappedItemProps[]

	/** Class for the list */
	className?: string

	/** any passthrough to render in the body of the list */
	children?: React.ReactNode

	/** Is this whole list in a loading state? Sets all list items to loading only if true. */
	isLoading: boolean
}

const List = (props: IListProps): React.ReactElement => {
	const {
		header,
		items,
		className,
		isSmall,
		areSeparatorsVisible: areSeparatorsVisibleProp,
		children,
		selectableType,
		isLoading
	} = props

	// seperators a true by default
	const areSeparatorsVisible =
		typeof areSeparatorsVisibleProp === 'boolean'
			? areSeparatorsVisibleProp
			: true

	const parentClass = cx('list', className, {
		'list-small': isSmall,
		'list--separators-hidden': !areSeparatorsVisible,
		'loading-placeholder': isLoading
	})

	return (
		<Fragment>
			{header && <ListHeader isSmall={isSmall} {...header} />}
			<ul className={parentClass}>
				{items &&
					items.map((item, idx) => {
						const listItem = item as IListItemProps
						const expandablListItem = item as IExpandableListItemProps

						if (listItem.title) {
							return (
								<ListItem
									key={listItem.id}
									selectableType={
										typeof listItem.selectableType === 'string'
											? listItem.selectableType
											: selectableType
									}
									{...listItem}
									isSeparatorVisible={
										typeof listItem.isSeparatorVisible === 'boolean'
											? listItem.isSeparatorVisible
											: areSeparatorsVisible
									}
								/>
							)
						}
						return <ExpandableListItem key={idx} {...expandablListItem} />
					})}
				{children && children}
			</ul>
		</Fragment>
	)
}

List.defaultProps = {
	header: null,
	className: '',
	isSmall: false,
	areSeparatorsVisible: true
}

export default List
