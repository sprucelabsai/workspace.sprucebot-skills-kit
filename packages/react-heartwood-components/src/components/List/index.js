export default List from './List'
export { default as ListHeader } from './components/ListHeader/ListHeader'
export { default as ListItem } from './components/ListItem/ListItem'
export { default as SortableList } from './components/SortableList/SortableList'
