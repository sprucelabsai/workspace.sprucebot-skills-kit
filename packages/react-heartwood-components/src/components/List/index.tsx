export { default as List, ListWrapper } from './List'
export { default as ListHeader } from './components/ListHeader/ListHeader'
export { default as ListItem } from './components/ListItem/ListItem'
export { default as SortableList } from './components/SortableList/SortableList'
export {
	default as ExpandableListItem
} from './components/ExpandableListItem/ExpandableListItem'
