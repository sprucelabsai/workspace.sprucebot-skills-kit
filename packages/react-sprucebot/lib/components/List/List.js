"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.ListWrapper = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _ListHeader = _interopRequireWildcard(require("./components/ListHeader/ListHeader"));

var _ListItem = _interopRequireWildcard(require("./components/ListItem/ListItem"));

var ListWrapper = function ListWrapper(props) {
  return _react.default.createElement("div", {
    className: "list-wrapper"
  }, props.children);
};

exports.ListWrapper = ListWrapper;

var List = function List(props) {
  var header = props.header,
      items = props.items,
      className = props.className,
      isSmall = props.isSmall;
  var parentClass = (0, _classnames.default)('list', className, {
    'list-small': isSmall
  });
  console.log({
    items: items
  });
  return _react.default.createElement(_react.Fragment, null, header && _react.default.createElement(_ListHeader.default, (0, _extends2.default)({
    isSmall: isSmall
  }, header)), _react.default.createElement("ul", {
    className: parentClass
  }, items.map(function (item, idx) {
    return _react.default.createElement(_ListItem.default, (0, _extends2.default)({
      key: idx
    }, item));
  })));
};

List.defaultProps = {
  header: null,
  className: '',
  isSmall: false
};
var _default = List;
exports.default = _default;