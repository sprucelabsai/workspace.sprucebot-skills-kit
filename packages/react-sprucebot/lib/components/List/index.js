"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ListHeader", {
  enumerable: true,
  get: function get() {
    return _ListHeader2.default;
  }
});
Object.defineProperty(exports, "ListItem", {
  enumerable: true,
  get: function get() {
    return _ListItem2.default;
  }
});
Object.defineProperty(exports, "SortableList", {
  enumerable: true,
  get: function get() {
    return _SortableList2.default;
  }
});
exports.default = void 0;

var _List = _interopRequireDefault(require("./List"));

var _ListHeader2 = _interopRequireDefault(require("./components/ListHeader/ListHeader"));

var _ListItem2 = _interopRequireDefault(require("./components/ListItem/ListItem"));

var _SortableList2 = _interopRequireDefault(require("./components/SortableList/SortableList"));

var _default = _List.default;
exports.default = _default;