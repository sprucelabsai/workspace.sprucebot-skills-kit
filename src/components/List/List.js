import styled from 'styled-components'
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Avatar from '../Avatar/Avatar'

export const List = styled.div.attrs({
	className: 'item__list'
})``

export class ListItem extends Component {
	render() {
		let {
			className,
			children,
			title,
			subtitle,
			rightInput,
			rightTitle,
			rightSubtitle,
			online,
			avatar,
			...props
		} = this.props

		// setup class name
		className = `${className || ''} item__list__item ${online ? '' : 'offline'}`

		// build children
		children = children || []
		if (!Array.isArray(children)) {
			children = [children]
		}

		// setup title/subtitle
		if (subtitle) {
			children.unshift(
				<div key="subtitle" className="sub__title">
					{subtitle}
				</div>
			)
		}

		if (title) {
			children.unshift(
				<div key="title" className="title">
					{title}
				</div>
			)
		}

		return (
			<div className={className} {...props}>
				{avatar && (
					<div className="avatar__outer__wrapper">
						{avatar === true ? (
							<Avatar online={online} />
						) : (
							<Avatar online={online} image={avatar} />
						)}
					</div>
				)}
				{children && <div className="item__details">{children}</div>}
				{(rightTitle || rightSubtitle || rightInput) && (
					<div className="content__right">
						{rightInput && rightInput}
						{rightTitle && <div className="title">{rightTitle}</div>}
						{rightSubtitle && <div className="sub__title">{rightSubtitle}</div>}
					</div>
				)}
			</div>
		)
	}
}

ListItem.propTypes = {
	friend: PropTypes.object,
	title: PropTypes.string,
	subtitle: PropTypes.string,
	rightInput: PropTypes.any,
	rightTitle: PropTypes.any,
	rightSubtitle: PropTypes.any,
	online: PropTypes.bool,
	avatar: PropTypes.oneOfType([PropTypes.string, PropTypes.bool])
}

ListItem.defaultProps = {
	online: true
}
