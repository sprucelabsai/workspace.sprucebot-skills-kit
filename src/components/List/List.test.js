import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { List, ListItem } from './List'

test('List renders', () => {
	const tree = renderer.create(<List />).toJSON()
	expect(tree).toMatchSnapshot()
})

test('ListItem should render non-arrays', () => {
	const tree = renderer.create(<ListItem>YOLO</ListItem>).toJSON()
	expect(tree).toMatchSnapshot()
})

test('List render send padding-bottom if its piled', () => {
	const tree = renderer.create(<List alignItems="flex-end" pile />).toJSON()
	expect(tree).toMatchSnapshot()
})

test('ListItem renders', () => {
	const tree = renderer.create(<ListItem />).toJSON()
	expect(tree).toMatchSnapshot()
})

test('ListItem should align items', () => {
	const tree = renderer.create(<ListItem alignItems="flex-end" />).toJSON()
	expect(tree).toMatchSnapshot()
})

test('ListItemAvatar should align where we specify', () => {
	const tree = renderer
		.create(<ListItem alignItems="flex-start" avatar />)
		.toJSON()
	expect(tree).toMatchSnapshot()
})

test('ItemRightContent should align where we specify', () => {
	const tree = renderer
		.create(<ListItem alignItems="flex-start" avatar />)
		.toJSON()
	expect(tree).toMatchSnapshot()
})

test('List should render title', () => {
	const tree = renderer
		.create(
			<List>
				<ListItem title={'foo'} />
			</List>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})

test('List should render subTitle', () => {
	const tree = renderer
		.create(
			<List>
				<ListItem title={'foo'} subtitle={'bar'} />
			</List>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})

test('List should render listitems and divs', () => {
	const tree = renderer
		.create(
			<List>
				<div>
					<ListItem title={'foo'} subtitle={'bar'} />
					<a>Im also a list item</a>
				</div>
			</List>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})

test('List should render ItemRightContent with a title', () => {
	const tree = renderer
		.create(
			<List>
				<div>
					<ListItem rightTitle={'foo'} subtitle={'bar'} />
					<a>Im also a list item</a>
				</div>
			</List>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
