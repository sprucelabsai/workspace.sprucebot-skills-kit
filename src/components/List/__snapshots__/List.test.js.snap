// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemRightContent should align where we specify 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 {
  position: relative;
  margin: 0 10px 0 0;
  border-radius: 50%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className=" ListItemWrapper item__list__item  c0"
>
  <div
    className="ItemAvatar avatar__outer__wrapper c1"
  >
    <div
      className="avatar__wrapper online "
    />
  </div>
  <div
    className="ItemDetail item__details c2"
  />
</div>
`;

exports[`List render send padding-bottom if its piled 1`] = `
.c0 {
  padding-bottom: 1.25em;
}

<div
  className="List item__list c0"
/>
`;

exports[`List renders 1`] = `
<div
  className="List item__list "
/>
`;

exports[`List should render ItemRightContent with a title 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 {
  text-align: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c4 {
  font-weight: 400;
}

.c2 {
  font-size: 0.75em;
}

<div
  className="List item__list "
>
  <div>
    <div
      className=" ListItemWrapper item__list__item  c0"
    >
      <div
        className="ItemDetail item__details c1"
      >
        <div
          className="ItemSubTitle sub__title c2"
        >
          bar
        </div>
      </div>
      <div
        className="ItemRightContent content__right c3"
      >
        <div
          className="ItemTitle title c4"
        >
          foo
        </div>
      </div>
    </div>
    <a>
      Im also a list item
    </a>
  </div>
</div>
`;

exports[`List should render listitems and divs 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 {
  font-weight: 500;
}

.c3 {
  font-size: 0.75em;
}

<div
  className="List item__list "
>
  <div>
    <div
      className=" ListItemWrapper item__list__item  c0"
      title="foo"
    >
      <div
        className="ItemDetail item__details c1"
      >
        <div
          className="ItemTitle title c2"
        >
          foo
        </div>
        <div
          className="ItemSubTitle sub__title c3"
        >
          bar
        </div>
      </div>
    </div>
    <a>
      Im also a list item
    </a>
  </div>
</div>
`;

exports[`List should render subTitle 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 {
  font-weight: 500;
}

.c3 {
  font-size: 0.75em;
}

<div
  className="List item__list "
>
  <div
    className=" ListItemWrapper item__list__item  c0"
    title="foo"
  >
    <div
      className="ItemDetail item__details c1"
    >
      <div
        className="ItemTitle title c2"
      >
        foo
      </div>
      <div
        className="ItemSubTitle sub__title c3"
      >
        bar
      </div>
    </div>
  </div>
</div>
`;

exports[`List should render title 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 {
  font-weight: 500;
}

<div
  className="List item__list "
>
  <div
    className=" ListItemWrapper item__list__item  c0"
    title="foo"
  >
    <div
      className="ItemDetail item__details c1"
    >
      <div
        className="ItemTitle title c2"
      >
        foo
      </div>
    </div>
  </div>
</div>
`;

exports[`ListItem renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className=" ListItemWrapper item__list__item  c0"
>
  <div
    className="ItemDetail item__details c1"
  />
</div>
`;

exports[`ListItem should align items 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className=" ListItemWrapper item__list__item  c0"
>
  <div
    className="ItemDetail item__details c1"
  />
</div>
`;

exports[`ListItem should render non-arrays 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className=" ListItemWrapper item__list__item  c0"
>
  <div
    className="ItemDetail item__details c1"
  >
    YOLO
  </div>
</div>
`;

exports[`ListItemAvatar should align where we specify 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 {
  position: relative;
  margin: 0 10px 0 0;
  border-radius: 50%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className=" ListItemWrapper item__list__item  c0"
>
  <div
    className="ItemAvatar avatar__outer__wrapper c1"
  >
    <div
      className="avatar__wrapper online "
    />
  </div>
  <div
    className="ItemDetail item__details c2"
  />
</div>
`;
